src/transformation/visitors/conditional.ts

```diff
@@ -27,32 +27,34 @@ function canBeFalsy(context: TransformationContext, type: ts.Type): boolean {
     }
 }
 
-function wrapInFunctionCall(expression: lua.Expression): lua.FunctionExpression {
-    const returnStatement = lua.createReturnStatement([expression]);
-
-    return lua.createFunctionExpression(
-        lua.createBlock([returnStatement]),
-        undefined,
-        undefined,
-        lua.FunctionExpressionFlags.Inline
-    );
-}
-
 function transformProtectedConditionalExpression(
     context: TransformationContext,
     expression: ts.ConditionalExpression
-): lua.CallExpression {
+): lua.Expression {
+    const tempVar = context.createTempForNode(expression.condition);
+
     const condition = context.transformExpression(expression.condition);
+
+    context.pushPrecedingStatements();
     const val1 = context.transformExpression(expression.whenTrue);
-    const val2 = context.transformExpression(expression.whenFalse);
+    const trueStatements = context.popPrecedingStatements();
+    trueStatements.push(lua.createAssignmentStatement(lua.cloneIdentifier(tempVar), val1, expression.whenTrue));
 
-    const val1Function = wrapInFunctionCall(val1);
-    const val2Function = wrapInFunctionCall(val2);
+    context.pushPrecedingStatements();
+    const val2 = context.transformExpression(expression.whenFalse);
+    const falseStatements = context.popPrecedingStatements();
+    falseStatements.push(lua.createAssignmentStatement(lua.cloneIdentifier(tempVar), val2, expression.whenFalse));
 
-    // (condition and (() => v1) or (() => v2))()
-    const conditionAnd = lua.createBinaryExpression(condition, val1Function, lua.SyntaxKind.AndOperator);
-    const orExpression = lua.createBinaryExpression(conditionAnd, val2Function, lua.SyntaxKind.OrOperator);
-    return lua.createCallExpression(orExpression, [], expression);
+    context.addPrecedingStatements([
+        lua.createVariableDeclarationStatement(tempVar, undefined, expression.condition),
+        lua.createIfStatement(
```

Comment:
Wouldn't this cause the condition and one of  the branches to be evaluated even if they should not be, if they are part of an expression like for example `const a = true || (a ? b : c)`?

Reply:
That wouldn't happen because the short-circuit logic for the or `||` opertaor would wrap the right side in another if clause. Here's the output of your example:
```lua
-- const x = true || (a ? b : c)
local ____temp_1 = true
if not ____temp_1 then
    local ____a_0
    if a then
        ____a_0 = b
    else
        ____a_0 = c
    end
    ____temp_1 = ____a_0
end
x = ____temp_1
```